GoogleSignIn/Tests/Unit/GIDVerifyAccountDetailTest.m

```diff
@@ -0,0 +1,144 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#if TARGET_OS_IOS
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDConfiguration.h"
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h"
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h"
+
+#import "GoogleSignIn/Tests/Unit/GIDFakeMainBundle.h"
+
+static NSString * const kClientId = @"FakeClientID";
+static NSString * const kServerClientId = @"FakeServerClientID";
+static NSString * const kOpenIDRealm = @"FakeRealm";
+static NSString * const kFakeHostedDomain = @"fakehosteddomain.com";
+
+@interface GIDVerifyAccountDetailTests : XCTestCase
+// The |UIViewController| object being tested.
+@property UIViewController *presentingViewController;
+
+// Fake [NSBundle mainBundle].
+@property GIDFakeMainBundle *fakeMainBundle;
+
+// The |GIDVerifyAccountDetail| object being tested.
+@property GIDVerifyAccountDetail *verifyAccountDetail;
+
+// The list of account details when testing [GIDVerifiableAccountDetail].
+@property NSArray<GIDVerifiableAccountDetail *> *verifiableAccountDetails;
+@end
+
+@implementation GIDVerifyAccountDetailTests
+
+#pragma mark - Lifecycle
+
+- (void)setUp {
+  [super setUp];
+
+  _presentingViewController = [[UIViewController alloc] init];
+
+  _verifyAccountDetail = [[GIDVerifyAccountDetail alloc] init];
+
+  GIDVerifiableAccountDetail *ageOver18Detail = [[GIDVerifiableAccountDetail alloc] initWithAccountDetailType:GIDAccountDetailTypeAgeOver18];
+  _verifiableAccountDetails = @[ageOver18Detail];
+
+  _fakeMainBundle = [[GIDFakeMainBundle alloc] init];
+}
+
+
+#pragma mark - Tests
+
+- (void)testInit {
+  [_fakeMainBundle startFakingWithClientID:kClientId];
+
+  GIDVerifyAccountDetail *verifyAccountDetail = [[GIDVerifyAccountDetail alloc] init];
+  XCTAssertNotNil(verifyAccountDetail.configuration);
+  XCTAssertEqual(verifyAccountDetail.configuration.clientID, kClientId);
+  XCTAssertNil(verifyAccountDetail.configuration.serverClientID);
+  XCTAssertNil(verifyAccountDetail.configuration.hostedDomain);
+  XCTAssertNil(verifyAccountDetail.configuration.openIDRealm);
+}
+
+- (void)testInit_noConfig {
+  [_fakeMainBundle startFakingWithClientID:kClientId];
+  [_fakeMainBundle fakeWithClientID:nil
+                     serverClientID:nil
+                       hostedDomain:nil
+                        openIDRealm:nil];
+  GIDVerifyAccountDetail *verifyAccountDetail = [[GIDVerifyAccountDetail alloc] init];
+
+  XCTAssertNil(verifyAccountDetail.configuration);
+}
+
+
+- (void)testInit_fullConfig {
+  [_fakeMainBundle startFakingWithClientID:kClientId];
```

Comment:
I think you would flip these two calls. Set up the fake with the data, and then tell the fake to start faking. It probably works both ways, but this way feels more logical to me.

Reply:
Seems that I get an error when I flip them. This is because `-[GIDFakeMainBundle startFakingWithClientID:]` initializes `NSMutableDictionary *_fakeConfig` and filling the fake with data before initializing will error since the `_fakeConfig` is null. I'll keep it as is unless there's another way to go about it.

Reply:
You could add an initializer that sets up the config. The initializer could take all of the parameters in the call starting on line 88 below. This would set up the config. Then, all that is needed is to call `[_fakeMainBundle startFaking]` (you could add a `-startFaking` method that doesn't take a client ID).

Reply:
Sounds good, I've implemented this suggestion.